etna/collections/models.py

```diff
@@ -244,14 +244,22 @@ def get_datalayer_data(self, request: HttpRequest) -> Dict[str, Any]:
 
     @cached_property
     def related_articles(self):
-        from etna.articles.models import ArticlePage
+        """Return a list of related pages for rendering in the related articles section
+        of the page. To add another page type, add it to the `exact_type` query below.
+        """
+
+        from wagtail.models import Page
+
+        from etna.articles.models import ArticlePage, FocusedArticlePage
 
         return (
-            ArticlePage.objects.exclude(pk=self.featured_article)
+            Page.objects.exclude(pk=self.featured_article.pk)
             .live()
             .public()
+            .exact_type(ArticlePage, FocusedArticlePage)
             .filter(pk__in=self.related_page_pks)
             .order_by("-first_published_at")
+            .specific()
             .select_related("teaser_image")
```

Comment:
I don't think this `select_related()` statement will work from here, since `Page` does not have a `teaser_image` field. It's probably best to remove it all together.

Reply:
Actually, if you wanted to keep things nice and efficient here, you could consider querying each model separately, sticking everything into a list, then sort that list in Python. I think you might see some performance gains by adding `prefetch_related("teaser_image__renditions")` to each query, too.

Reply:
Okay! I wasn't sure which was better (if you see the commit history you'll see I've re-done it 3 times 😆 )
If that's the best then I'll do that :)